### ISSUE TITLE
Craft 4 support?

### ISSUE THREAD

**Issue by timeverts** [kayqq/craft-cloudfront-purge#5]
Hi @kayqq and @simeon-smith,

Do you have any plans to port this plugin to Craft 4?

Thanks.

**Comment by ctangney-tulip** [comment]
@timeverts @kayqq I've updated the code for use with Craft 4. 

Shall I open a PR, or would it be more advisable to fork this? Not certain what your capacity for maintaining this is @kayqq

**Comment by nabilfreeman** [comment]
@timeverts @ctangney-tulip +1 for an update to this 😇

If you would be so kind, I think even just a PR would very useful for those of us comfortable with patches!

**Comment by timeverts** [comment]
Great work @ctangney-tulip. 

@kayqq and @simeon-smith, are you open and able to merge in a pull request and release a new version of the plugin? Or do you prefer that @ctangney-tulip release a brand new Craft 4 plugin based on the updated supporting code?

**Comment by ctangney-tulip** [comment]
@timeverts @nabilfreeman I have not yet heard back from @kayqq regarding his preference on maintaining this code.

Our use-case required use to invalidate multiple versions of the path on entry save, so I have also added basic support for invalidation of patterns, based on the entry being saved. An example of this would be something like:
- `/the-entry`
- `/the-entry/`

More than happy to open a PR for this! If necessary, I can also fork this repository such that the code is available. Perhaps @simeon-smith has some input on this?

**Comment by ctangney-tulip** [comment]
Completing the circle of references: https://github.com/kayqq/craft-cloudfront-purge/pull/6

**Comment by kayqq** [comment]
I dont have capacity anymore to support this plugin but am happy to add new owners to this. 

I merged #6 in. Hopefully that resolves your issue @ctangney-tulip @nabilfreeman @timeverts

**Comment by timeverts** [comment]
Thanks for providing an update @kayqq.

The merge into master was helpful, but to be honest, there's a bit more to do to release this as a [Craft 4 plugin](https://craftcms.com/docs/4.x/extend/plugin-guide.html) into the plugin store. 

@ctangney-tulip, if you're familiar with the process to create a new plugin, and if @kayqq is happy to [transfer ownership of the repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/transferring-a-repository#transferring-a-repository-owned-by-your-personal-account) to you, I'd suggest that. Then you can make the final changes and release it to the Craft 4 plugin store.

**Comment by nabilfreeman** [comment]
Thank you all, it's so encouraging to see speedy action on projects like this! Really genuinely appreciate it 👏👏👏👏👏👏

**Comment by ctangney-tulip** [comment]
@timeverts agreed -- still a good bit to do in order to get this published. This was ultimately a quick / dirty hot-fix to get things working for our needs.

I will discuss with my team the possibility of taking ownership of this plugin. Our bandwidth is limited for the moment, but it might be something I can attempt to support in a personal capacity.

I do not use Craft in my personal projects, nor do I have access to an instance outside of my organization. This would ultimately make personal ownership a little tricky for me to sign up for 😞

**Comment by kayqq** [comment]
Should be on the Craft 4 store now

**Comment by nabilfreeman** [comment]
@kayqq LEGEND 🚀 🎉 🎊

**Comment by ctangney-tulip** [comment]
@kayqq do you have an email that would work for me to contact you at with some questions regarding ownership of this codebase?
